### ISSUE TITLE
Environmental declarations not working in functions

### ISSUE THREAD

**Issue by dickvand** [kobbled/tp_plus#39]
Environmental file:
```
reg1 := R[1]
```

Function declaration:
```
def foo()
  reg1 = 1
end
```

Results in this error:
```
C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:214:in `rescue in eval': Runtime error on line 1: (RuntimeError)
Variable (reg1) not defined
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:177:in `eval'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/function.rb:165:in `output_program'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:121:in `block in output_functions'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:120:in `each'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:120:in `output_functions'
        from C:/Users/184/TP-plus/tp_plus/bin/tpp:128:in `<main>'
C:/Users/184/TP-plus/tp_plus/lib/tp_plus/base_block.rb:123:in `get_var': Variable (reg1) not defined (RuntimeError)
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/nodes/var_node.rb:10:in `target_node'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/nodes/var_node.rb:35:in `eval'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/nodes/assignment_node.rb:44:in `identifier_string'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/nodes/assignment_node.rb:55:in `eval'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:199:in `block in eval'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:195:in `each'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:195:in `eval'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/function.rb:165:in `output_program'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:121:in `block in output_functions'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:120:in `each'
        from C:/Users/184/TP-plus/tp_plus/lib/tp_plus/interpreter.rb:120:in `output_functions'
        from C:/Users/184/TP-plus/tp_plus/bin/tpp:128:in `<main>'
```

**Comment by kobbled** [comment]
You have to pass anything that you need external to the function into the function scope. Same for namespaces. If you are using an environment file you need to declare `using env`

```
def foo()
  using env
  reg1 = 1
end
```

If its a namespace you have to scope it through the namespace into the function
```
reg1 := R[1]

namespace Space1
  using reg1

  def foo()
    using reg1
  end
end

```

If you arn't using an environment file, and don't want to say "using ..." for dozens of registers, group them in a namespace, and pass in the namespace. You can also swap namespaces in functions that have the exact same members. However this is only really practical  at this point when the preprocessor is introduced. 

Doing it this way gives you control as to what the namespace, or functions have access to. It is also more comprehensible when you have dozens of files with dependencies on each other. It was also 10x easier to program this way into the existing interpreter.

see: https://github.com/kobbled/tp_plus/blob/master/examples.md#namespace-scoping, https://github.com/kobbled/tp_plus/blob/master/examples.md#imports

**Comment by kobbled** [comment]
I should note, that the vscode extension can handle includes/imports if you put this in the package.json file:
``` package.json
"includes" : ["../frames", "../poses"]
```
I've updated the readme a bit https://github.com/kobbled/vscode-tpp-extension.

The string in the includes is the relative path of the folder with the tpp in it.

```TP+
import frame, pose
```
is the name of the .tpp file you want to import. I've excluded the .tpp at the end, but it might be more understandable with it in.

Then any namespaces or functions are declared with:

```TP+
using FameNS, PoseNS
```

So its **includes** : folder level (*package.json* file), **import** : file level (*.tpp* file), and **using** : namespace/function level (*.tpp* file).
